vdm-io/Joomla-Component-Builder
# Adding Expansion Development Method and more...

Pull Request for many Issue

### Summary of Changes
- Added a switch to control the message display of the expand feature when cronjob runs
- Added the feature that allows you to setup the behaviour of a selected set of components that will be auto build and installed. So you do not need to manually compile and install those components any more, the system does all that automatically for you. We have called the feature Expansion Development Method, and it can be set in the global settings of JCB under the Development Method tab.
- Fixed the auto backup system, to allow JCB to be backed up automatically
- Tweaked the field database area to insure that users are forced to set the database values
- Fixed a strange mismatch issue with JCB package import with more then 10 components at a time. Also fixed the translation entranslation to source.
- Improved JCB to now be able to have other languages then en-GB as its base/main/global language. This will allow you to build your components in a language of your choice.
- Added edit links to templates and layouts.
- Fixed the alias and title selection issue in admin fields area.
- Added all available languages in Joomla to language area.
- Added minify of JS to the compiler area, also improved the infusion area to make more use of the getCustomScriptBuilder method
- Fixed the Add JavaScript (views-footer) option, since it did not add the javascript, but now does. Also added some spinner to the Joomla Components view.
- Added the option to clone a Component and all its linked data, also to import a JCB package without merging with local Component
- Keeping naming consistency resolved gh-265. Started adding the needed features for the new translation improvement gh-261
- Added the php field properties to a textarea outside of the subform for easier editing

### Testing Instructions
- Check that the demo component is able to compile and that you can all over JCB edit, create and all the other normal functionality still works as expected.
- Check that you can import the demo component without merging, and so have duplicates of all fields, views and other linked data to the component.
-  Test drive JCB with other language as en-GB being the main language.
- Check that the auto back-up system still works.
- Test the Expansion Development Method, it can be set in the global settings of JCB under the Development Method tab.

### Expected result
- That JCB is stable and now having more features and improvements then before.
